lib/FrontendTool/FrontendTool.cpp

```diff
@@ -881,7 +881,7 @@ static bool performCompileStepsPostSILGen(
 /// \param Instance Will be reset after performIRGeneration when the verifier
 ///                 mode is NoVerify and there were no errors.
 /// \returns true on error
-static bool performCompile(CompilerInstance &Instance,
+bool swift::performCompile(CompilerInstance &Instance,
```

Comment:
This was the part I was most worried about. Not sure what you'd prefer I do about it. I can move big chunks of FrontendTool.cpp to Frontend/ but that seems like it might be suboptimal. But maybe necessary if we want LLDB and friends to be able to invoke this without linking to libFrontendTool.a?

Also: a fair amount of `performCompile` isn't strictly necessary here: it handles all possible Invocation cases, most of which we're not doing. I can copy-out the minimal steps to run just this one thing (possibly sinking bits of common logic into methods on `CompilerInstance`) if that seems more suitable?

Reply:
I definitely don't want to duplicate `performCompile`, even with large chunks of it cut out, but if it can be factored nicely…